app/routes/index.js

```diff
@@ -310,39 +265,86 @@ const storage = multer.diskStorage({
 			else
 				res.redirect('/admin');
 		});
-	app.post('/adduser', upload.single('profile_pic'), async function(req,res){
+	app.route('/adduser').post(adminAuth, async function(req,res){
 		try {
-			const user = new User(req.body);
-			console.log(user)
-			if(req.file){
-				user.profile_pic = req.file.tempFilePath;
-			}
-			await user.save()
-			res.json("User added successfully");
+			const form = new formidable.IncomingForm();
+
+    	form.parse(req, async function(err, fields, files){
+		console.log(fields)
+		console.log(files)
+		
+		const user = new User(fields);
+        var oldPath = files.profile_pic.filepath;
+		console.log(__dirname)
+        var newPath = path1.join(__dirname, '/../../upload')+ '/'+fields.username
+        var rawData = fs.readFileSync(oldPath)
+        fs.writeFile(newPath, rawData, function(err){
+            if(err){console.log(err); return res.json({err: true})}
+            
+        })
+		user.profile_pic=newPath;
+		console.log(user)
+		await user.save()
+		res.json({err: false, msg: "Successfully uploaded user"})
+  })
 		} catch (error) {
 			console.log("error");
 			res.json(error);
 			
 		}
 		
 	});
+	app.route('/adduser').get(adminAuth, async function(req,res){
+		const users=await User.find()
+		res.json(users)
+	})
 	app.route('/addevent').
-	post(adminAuth, async function(req,res){
+	post(async function(req,res){
 		try {
-			const event = new Event(req.body);
-			event.save()
-			res.json(event);
+			const form = new formidable.IncomingForm({
+				multiples: true,
+				keepExtensions: true,
+			  });
+			console.log(form)
+    		form.parse(req, async function(err, fields, files){
+			console.log(files)
+			const event = new Event()
+			event.name=fields.name
+			event.description=fields.description
+			event.date=Date.parse(fields.date)
+			let sz= files.gallery.length
+			console.log("size is ", sz)
+			for(let i=0;i<sz;i++)
```

Comment:
Also you can use javascript's `forEach` method. It looks more elegant and alleviates worries about sizes and lengths :smile: